packages/components/view-switcher/src/view-switcher.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  Children,
+  isValidElement,
+  cloneElement,
+  useState,
+  useEffect,
+  type ReactNode,
+  type ReactElement,
+} from 'react';
+import { css } from '@emotion/react';
+import ViewSwitcherButton from './view-switcher-button';
+import { warning } from '@commercetools-uikit/utils';
+
+type TReactChild = {
+  type?: { displayName: string };
+} & ReactElement;
+
+export type TViewSwitcherProps = {
+  /**
+   * Indicates that the view switcher can be reduced to save space
+   */
+  isCondensed?: boolean;
+  /**
+   * Pass one or more `ViewSwitcher.Button` components
+   */
+  children: ReactNode;
+  /**
+   * Will be triggered whenever a `ViewSwitcher.Button` is selected. Called with the ViewSwitcherButton value
+   */
+  onChange?: (value: string) => void;
+  /**
+   * Indicates the default selected button
+   */
+  defaultSelected: string;
+};
+
+const ViewSwitcher = (props: TViewSwitcherProps) => {
+  const [selectedButton, setSelectedButton] = useState<string>(
+    props.defaultSelected
+  );
+
+  useEffect(() => {
+    const childrenAsArray = Children.toArray(props.children) as TReactChild[];
+    const optionChildrenAsArray = childrenAsArray.filter(
+      (child) => child.type.displayName === ViewSwitcherButton.displayName
+    );
```

Comment:
I'm not sure if I'm missing something here because, since we don't expect the `ViewSwitcher` to have any other component than `ViewSwitcherButton`, perhaps we could avoid this filtering iteration. I mean, it seems it does not make sense to provide any other type of children, right?

Also, we use the hook because we expect the children to change dynamically? (perhaps some use case you have in mind)

Reply:
It's meant to be a check in case the component wasn't used properly. But, I think it makes sense to remove the filter and hook.

Reply:
It has been removed in this commit: c7f430ef5530845eb75c43525902691340a711ef